js/src/carousel.js

```diff
@@ -230,23 +261,74 @@ class Carousel {
       $(this._element)
         .on(Event.MOUSEENTER, (event) => this.pause(event))
         .on(Event.MOUSELEAVE, (event) => this.cycle(event))
-      if ('ontouchstart' in document.documentElement) {
+    }
+
+    this._addTouchEventListeners()
+  }
+
+  _addTouchEventListeners() {
+    if (!this._touchSupported) {
+      return
+    }
+
+    const start = (event) => {
+      event.preventDefault()
+
+      if (this._pointerEvent) {
+        this.touchStartX = event.originalEvent.clientX
```

Comment:
ideally also check first that the pointer event is of `pointerType=="touch"` - and possibly even `pointerType="pen"` if we want to allow pen/stylus swiping - as otherwise the swipe will also happen for mouse users. see https://w3c.github.io/pointerevents/#dom-pointerevent-pointertype